### PR TITLE
Restaurants page WIP

### PR DESCRIPTION
**Pull request by Lordnickim**
Created Menu Page, Menu Item Page, Restaurant Page 
Added MenuItem Compoenet, Incrementor component, Item Containers
Added 3 Reducers: ItemOrder, CurrentItem, AllOrderItems.

### CODE REVIEW

File: src/client/state/actions/allActions.js [hunk 1]
```diff
@@ -1,7 +1,23 @@
 import { addName, addUserName, setNumber } from '../reducers/namesReducer';
+import { addItem, subtractItem, UpdateItemPrice, UpdateItemDescription, resetItemQuantity } from '../reducers/menuItemReducer';
+import { UpdateItemName, UpdateItemId, UpdateItemQuantity, UpdateTotalPrice, UpdateRestaurantId } from '../reducers/orderItemReducer';
+import { addItemToOrders, addToPriceTotal } from '../reducers/allOrderItemsReducer';
 
 const allActions = {
-  addName, addEmail
+  addName,
+  // addEmail,
```
Comment (Lordnickim) [inline src/client/state/actions/allActions.js:8]: Ran into an issue where addEmail wasnt imported so commented it out for now until login page updates it?

Comment (scottiehansen) [reply src/client/state/actions/allActions.js:8]: Yeah, my bad, I added that action and forgot to edit it out before I merged earlier.

**Review by scottiehansen — APPROVED** [review]
Everything looks good to me (I didn't review the redux stuff, I don't think I'm fluent enough in it to spot a problem)

File: src/client/components/MenuItemIncrementor.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  function addHandler () {
+    dispatch(allActions.addItem())
+    dispatch(allActions.UpdateItemQuantity(count+1));
+  }
+
+  return(
+    <MainConatiner>
+      <CirleButton onClick={()=>subtractHandler()}>-</CirleButton>
```
Comment (erikeh) [inline src/client/components/MenuItemIncrementor.jsx:35]: fyi, you dont need to create an anonymous function to place the handler function if you dont need to give it an event object.

onClick={subtractHandler}  // is better if you dont need the event object
onClick={e => subtractHandler(e)} // anonymous function if you need the event object

File: src/client/components/MenuPage.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  const dispatch = useAppDispatch();
+  const history = useHistory();
+
+  function clickHandler (entry) {
+    //reroute
+    console.log(entry.name)
+    console.log(entry.price)
+    dispatch(allActions.UpdateItemName(entry.name));
+    dispatch(allActions.UpdateItemPrice(entry.price));
+    dispatch(allActions.UpdateItemDescription(entry.description));
```
Comment (erikeh) [inline src/client/components/MenuPage.jsx:34]: would highly recommend destructuring the individual actions from the allActions to keep code dry!

File: src/client/components/MenuItemContainer.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { useAppDispatch, useAppSelector } from '../state/hooks';
+import { Button } from 'react-bulma-components';
```
Comment (erikeh) [inline src/client/components/MenuItemContainer.jsx:3]: Lets start using the shared component 'OrangeButton' sooner than later, just to make sure that everything is compatible and all good

File: src/client/components/MenuItemPage.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  function clickHandler () {
+    console.log(totalPrice)
+
+    dispatch(allActions.addToPriceTotal(totalPrice));
+    dispatch(allActions.addItemToOrders(currentOrder));
+    //add user id?
+    history.push("/Menu");
+  }
+
+  React.useEffect(()=>{
```
Comment (erikeh) [inline src/client/components/MenuItemPage.jsx:39]: lol remember to keep the code consistent!

**Review by erikeh — APPROVED** [review]

File: src/client/components/MenuItemContainer.jsx (hunk 4 above)
Comment (Lordnickim) [reply src/client/components/MenuItemContainer.jsx:3]: gotcha think that bulma button was just something i forgot to take out lols

**Pull request merged**